### PR TITLE
chore(pkglint): validate that package `maturity` is correct

### PR DESCRIPTION
**Pull request by skinny85**
We were missing a few cases where we did not validate that a package had the correct `maturity` set:

- A package with L2s could still have a `maturity` of 'cfn-only'
- A package with only L1s could still have a non-'cfn-only' `maturity`

Fix that in the linter, along with the violations discovered.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: tools/pkglint/lib/rules.ts [hunk 1]
```diff
@@ -284,8 +284,32 @@ export class MaturitySetting extends ValidationRule {
       maturity = 'deprecated';
     }
 
+    const packageLevels = this.determinePackageLevels(pkg);
+
+    const hasL1s = packageLevels.findIndex(level => level === 'l1') !== -1;
+    const hasL2s = packageLevels.findIndex(level => level === 'l2') !== -1;
```
Comment (jogold) [inline tools/pkglint/lib/rules.ts:290]:
```suggestion
    const hasL1s = packageLevels.some(level => level === 'l1');
    const hasL2s = packageLevels.some(level => level === 'l2');
```

File: packages/@aws-cdk/aws-acmpca/package.json [hunk 2]
```diff
@@ -92,7 +92,7 @@
     "node": ">= 10.13.0 <13 || >=13.7.0"
   },
   "stability": "experimental",
-  "maturity": "cfn-only",
```
Comment (NetaNir) [inline packages/@aws-cdk/aws-acmpca/package.json:95]: Why?
Looks like it only have an [l1](https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/aws-acmpca)

Comment (nija-at) [reply packages/@aws-cdk/aws-acmpca/package.json:95]: Not sure if this qualifies as an L2 - https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-acmpca/lib/certificate-authority.ts

Comment (skinny85) [reply packages/@aws-cdk/aws-acmpca/package.json:95]: I think the `certificate-authority.ts` definitely makes it an L2.

**Review by NetaNir — APPROVED** [review]

**Pull request merged**